File: src/main/java/com/fast/backend/vehicle/domain/VehicleSource.java
package com.fast.backend.vehicle.domain;

/**
 * 차량 데이터의 출처. 실물 지게차인지 Isaac Sim 시뮬레이션인지 구분한다.
 * vehicleId 명명 규칙(예: REAL-F01, SIM-F01)의 최종 확정과는 무관하게, 이 값 자체는
 * FR-501-1 범위에서 지금 확정해 구현한다(prompt16.md 4장).
 */
public enum VehicleSource {
    REAL,
    SIMULATION
}

File: frontend/monitoring-control-system/lib/monitoring/vehicleSource.ts
export type VehicleSource = "real" | "sim" | "unknown"

/**
 * 백엔드 정규 ID(SIM-F01)와 Isaac 외부 ID(sim01)를 같은 시뮬 차량으로 분류한다.
 * 하이픈·언더스코어 유무와 대소문자만 허용하고, SIM-FLOW 같은 임의 접두어는 시뮬로 오인하지 않는다.
 */
export function resolveVehicleSource(vehicleId: string): VehicleSource {
  const compactId = vehicleId.trim().toUpperCase().replace(/[-_]/g, "")
  if (/^REALF?\d+$/.test(compactId)) return "real"
  if (/^SIMF?\d+$/.test(compactId)) return "sim"
  return "unknown"
}
